File: examples/java/wrist-tempo/app/src/main/java/com/example/wrist_tempo/MainActivity.java
package com.example.wrist_tempo;

import android.app.Activity;
import android.os.Bundle;
import android.widget.TextView;

/**
 * Wear OS entry point for wrist-tempo (legacy Android Wear).
 *
 * <p>Renders res/layout/activity_main.xml, whose root is an
 * androidx.wear.widget.BoxInsetLayout — the maintained view-based
 * layout for round watch faces (WearableActivity is deprecated).</p>
 *
 * <p>THE TWO BUILDS: this file needs the Android SDK and the
 * androidx.wear AAR, so it is built with Android tooling
 * (Android Studio / Gradle) — NOT by the Maven build, which covers
 * only the pure logic in src/main/java/. When assembling the Android
 * build, add src/main/java/ as a source root so this activity can
 * resolve {@link Greeting}. See the README section "The two builds".</p>
 */
public class MainActivity extends Activity {

    @Override
    protected void onCreate(final Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        final TextView greetingView = findViewById(R.id.greeting);
        greetingView.setText(Greeting.greet("wrist-tempo"));
    }
}

File: examples/kotlin/wrist-counter/app/src/main/kotlin/com/example/wrist_counter/MainActivity.kt
package com.example.wrist_counter

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.wear.compose.material.MaterialTheme
import androidx.wear.compose.material.Text

/** Returns the greeting assembled from the project name. */
fun greeting(projectName: String): String = "Hello from $projectName!"

/** Wear OS entry point hosting the Compose UI. */
class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContent { WearApp() }
    }
}

/** Root composable: centers the greeting on the round watch face. */
@Composable
fun WearApp() {
    MaterialTheme {
        Box(
            modifier = Modifier.fillMaxSize(),
            contentAlignment = Alignment.Center,
        ) {
            Text(text = greeting("wrist-counter"))
        }
    }
}
